lib/compilers/manual-solutions-compiler.ts

```diff
@@ -0,0 +1,62 @@
+import * as fs from "fs";
+import * as path from "path";
+import tmp from "tmp";
+import Course from "../models/course";
+import Language from "../models/language";
+import { glob } from "glob";
+
+class ManualSolutionsCompiler {
+  course: Course;
+
+  constructor(course: Course) {
+    this.course = course;
+  }
+
+  async compileAll() {
+    const languages = this.languages();
+
+    for (const language of languages) {
+      await this.compileForLanguage(language);
+    }
+  }
+
+  async compileForLanguage(language: Language) {
+    console.log(`- compiling manual solutions for ${this.course.slug}-${language.slug}`);
+
+    const stages = this.course.stages;
+    const firstStage = stages[0];
+    const firstStageCodeDirectory = path.join(this.course.solutionsDir, language.slug, firstStage.solutionDir, "code");
+    const firstStageFiles = await glob("**/*", {
```

Comment:
```suggestion
    const firstStageFilePaths = await glob("**/*", {
```

("files" can be confused for file objects)